builtInFunctions/esdtDataStorage.go

```diff
@@ -411,8 +414,21 @@ func (e *esdtDataStorage) SaveESDTNFTToken(
 	return marshaledData, acnt.AccountDataHandler().SaveKeyValue(esdtNFTTokenKey, marshaledData)
 }
 
+func (e *esdtDataStorage) checkTheCorrectTokenTypeIsSet(esdtData *esdt.ESDigitalToken) bool {
+	if !e.enableEpochsHandler.IsFlagEnabled(DynamicEsdtFlag) {
+		return true
+	}
+
+	if esdtData.Type != uint32(core.NonFungible) {
+		// it means that the token type was already migrated.
+		return true
+	}
+
+	return false
+}
+
 // removeNFTMetadataFromSystemAccountIfNeeded removes the NFT metadata from the system account if needed
-func (e *esdtDataStorage) removeNFTMetadataFromSystemAccountIfNeeded(esdtTokenKey []byte, nonce uint64, esdtData *esdt.ESDigitalToken) error {
+func (e *esdtDataStorage) migrateTokenTypeAndMetadataIfNeeded(esdtTokenKey []byte, nonce uint64, esdtData *esdt.ESDigitalToken) error {
```

Comment:
update or remove comment

Reply:
+1

Reply:
Updated the comment. It now states the reason why this function is needed.